htdocs/core/lib/security.lib.php

```diff
@@ -247,6 +248,25 @@ function restrictedArea($user, $features, $objectid = 0, $tableandshare = '', $f
 		{
 			if (! $user->rights->projet->lire && ! $user->rights->projet->all->lire) { $readok=0; $nbko++; }
 		}
+		elseif ($feature == 'agenda')
```

Comment:
This part of code is dedicated to check on brut permission only, whatever is object analyzed. Few lines later, there is a call to 
checkUserAccessToObject()
If we must make a test on a dedicated object id, we must add it in this function checkUserAccessToObject .
Can you try to move this code into this function checkUserAccessToObject()